packages/code-block/partials/clipboard-button/index.tsx

```diff
@@ -41,14 +47,14 @@ function ClipboardButton({ className, getText }) {
   useEffect(() => {
     // Clear any pending timeouts, which can occur if the
     // button is quickly clicked multiple times
-    clearTimeout(resetTimeout)
+    window.clearTimeout(resetTimeout)
     // Only run the copiedState reset if it's needed
     const needsReset = copiedState != null
     if (needsReset) {
       // Let failure messages linger a bit longer
       const resetDelay = copiedState == false ? 4000 : 1750
       // Set the timeout to reset the copy success state
-      setResetTimeout(setTimeout(() => setCopiedState(null), resetDelay))
+      setResetTimeout(window.setTimeout(() => setCopiedState(null), resetDelay))
```

Comment:
Curious, why the need to use the `window.` prefix here?

Reply:
It's a node/dom thing in TypeScript. At least how we have it configured, `setTimeout` returns a `NodeJS.Timeout`, whereas `clearTimeout` accepts a `number`. By using the `window.` prefix, we are able to get a reference to the dom versions of those APIs. Since this is being called in a `useEffect`, it should be safe to rely on `window`.